lib/net.js

```diff
@@ -223,7 +224,9 @@ function Socket(options) {
   else if (options === undefined)
     options = {};
 
-  stream.Duplex.call(this, options);
+  // There is already a "no-half-open enforcer" in place so there is no need to
+  // inherit it again from `Duplex`.
```

Comment:
Can you please add a note along the following lines: `Socket is still inheriting from Duplex, but we are just using a slimmed down constructor. socket instanceOf Duplex would keep returning true`

Reply:
Will do.

Reply:
@mcollina updated, PTAL.